www/guides/react.md

```diff
@@ -0,0 +1,361 @@
+
+Open up your terminal and head to a directory where you want to put your new project. Now run `npx create-snowpack-project react-snowpack --template @snowpack/project-template-minimal`. This creates a new directory called `react-snowpack` and installs the `@snowpack/project-template-minimal` inside it.
+
+You can now head to that new `react-snowpack` directory with `cd react-snowpack`. And to start up Snowpack run `npm run start`. You should see your new website up and running!
+
+> 💡 Tip: the `README.md` in your new project contains useful information about what each file does.
+
+Image: screenshot of project-template-minimal, which shows "Hello world" in text on a white background.
+
+> 💡 Tip: you can also use Yarn for any of the commands in this tutorial.
+
+Now that you have a basic project up and running, it's time to install React. To install React, run the following command in your project directory:
+
+```bash
+npm install react react-dom --save
+```
+
+> 💡 Tip: in this tutorial commands use `--save` for frontend libraries and `--save-dev` for your developer tools to keep them organized separately. This is totally up to you, though.
+
+## Create your first React component
+
+Snowpack has built in support for JSX in files with the `.jsx` extension. That means that there's no additional plugins or configuration needed to write your first React component, but you will need to rename your `index.js` file to `index.jsx` to take advantage of this.
+
+```bash
+mv index.js index.jsx
+```
+
+> 💡 Tip: You do not need to update your `index.html` script tag to point to `index.jsx`. Because JSX isn't supported in the browser, Snowpack builds your file to a runnable `.js` file. This is good to remember when you're referencing built files in HTML `<script src="">` and `<link href="">` elements.
+
+You can now import React in `index.jsx` and add a simple test component just to make sure it's working:
+
+```diff
+/* Add JavaScript code here! */
+- console.log('Hello World! You did! Welcome to Snowpack :D');
++import React from 'react';
++import ReactDOM from 'react-dom';
++ReactDOM.render(<div>"HELLO REACT"</div>, document.getElementById('root'));
+```
+
+Since the React code is rendering into an element with the ID `root`, we'll need to add that to `index.html`:
+
+```diff
+  <body>
+-    <h1>Welcome to Snowpack!</h1>
++    <div id="root"></div>
+    <script type="module" src="/index.js"></script>
+```
+
+Image: screenshot of the project, which shows "HELLO REACT" on a white background
+
+You've just created your first React component in Snowpack! And also gotten your project prepared for building more with React.
+
+## Customizing your file layout
+
+If you're going to be adding a bunch of new files, you probably don't want them in the root directory. Snowpack is flexible enough to support whatever project layout that you prefer. For this guide, let's use a popular project layout from the React community.
```

Comment:
New user: Why not?

maybe instead: "you probably don't want them crowding up your top-level root directly."